vaadin-grid-flow-parent/vaadin-grid-flow/src/test/java/com/vaadin/flow/component/grid/GridDataProviderConstructorTest.java

```diff
@@ -0,0 +1,354 @@
+                    "Dev"));
+
+    @Test
+    public void constructorBackEndDataProvider() {
+        Grid<Person> gridBackendDataProvider = createGrid(
+                new PersonBackendDataProvider());
+        Assert.assertEquals(people.get(0), gridBackendDataProvider.getLazyDataView().getItem(0));
+        Assert.assertEquals(people.get(1), gridBackendDataProvider.getLazyDataView().getItem(1));
+        Assert.assertEquals(people.get(2), gridBackendDataProvider.getLazyDataView().getItem(2));
+        Assert.assertEquals(people.get(3), gridBackendDataProvider.getLazyDataView().getItem(3));
+    }
+
+    @Test
+    public void constructorGenericDataProvider() {
+        Grid<Person> gridGenericDataProvider = createGrid(
+                new PersonGenericDataProvider());
+        Assert.assertEquals(people.get(0), gridGenericDataProvider.getGenericDataView().getItem(0));
+        Assert.assertEquals(people.get(1), gridGenericDataProvider.getGenericDataView().getItem(1));
+        Assert.assertEquals(people.get(2), gridGenericDataProvider.getGenericDataView().getItem(2));
+        Assert.assertEquals(people.get(3), gridGenericDataProvider.getGenericDataView().getItem(3));
+    }
+
+    @Test
+    public void constructorInMemoryDataProvider() {
+        Grid<Person> gridInMemoryDataProvider = createGrid(
+                new PersonInMemoryDataProvider());
+        Assert.assertEquals(people.get(0), gridInMemoryDataProvider.getGenericDataView().getItem(0));
+        Assert.assertEquals(people.get(1), gridInMemoryDataProvider.getGenericDataView().getItem(1));
+        Assert.assertEquals(people.get(2), gridInMemoryDataProvider.getGenericDataView().getItem(2));
+        Assert.assertEquals(people.get(3), gridInMemoryDataProvider.getGenericDataView().getItem(3));
+    }
+
+    @Test
+    public void constructorListDataProvider() {
+        Grid<Person> gridListDataProvider = createGrid(
+                new PersonListDataProvider());
+        Assert.assertEquals(people.get(0), gridListDataProvider.getListDataView().getItem(0));
+        Assert.assertEquals(people.get(1), gridListDataProvider.getListDataView().getItem(1));
+        Assert.assertEquals(people.get(2), gridListDataProvider.getListDataView().getItem(2));
+        Assert.assertEquals(people.get(3), gridListDataProvider.getListDataView().getItem(3));
+    }
+
+    @Test
+    public void constructorCollection() {
+        Grid<Person> gridCollection = createGrid(people);
+        Assert.assertEquals(people.get(0), gridCollection.getGenericDataView().getItem(0));
+        Assert.assertEquals(people.get(1), gridCollection.getGenericDataView().getItem(1));
+        Assert.assertEquals(people.get(2), gridCollection.getGenericDataView().getItem(2));
+        Assert.assertEquals(people.get(3), gridCollection.getGenericDataView().getItem(3));
+    }
+
+    private final Grid<Person> createGrid(DataProvider dataProvider) {
+        Grid<Person> grid = new Grid<>(dataProvider);
```

Comment:
The constructor usage here is problematic, because it will effectively only test a single constructor (`public Grid(DataProvider<T, Void> dataProvider)`) for all cases.

Grids should be constructed in their respective test cases with passing the specific type of data provider, rather then the general `DataProvider` type.